d3/d3.d.ts

```diff
@@ -74,9 +74,9 @@ declare module d3 {
              * Derive an attribute value for each node in the selection based on bound data.
              *
              * @param name The attribute name, optionally prefixed.
-             * @param value The function of the datum (the bound data item) and index (the position in the subgrouping) which computes the attribute value. If the function returns null, the attribute is removed.
+             * @param value The function of the datum (the bound data item), index (the position in the subgrouping), and inner index (overall position in nested selections) which computes the attribute value. If the function returns null, the attribute is removed.
              */
-            attr(name: string, value: (datum: Datum, index: number) => Primitive): Update<Datum>;
+            attr(name: string, value: (datum: Datum, index: number, innerIndex?: number) => Primitive): Update<Datum>;
```

Comment:
Based on http://bost.ocks.org/mike/nest/, it looks like the last argument should be "`outerIndex`" instead of "`innerIndex`".